### PR TITLE
Mark as obsolete early Node 12

### PR DESCRIPTION
**Pull request by zloirock**

### CODE REVIEW

File: environments.json [hunk 1]
```diff
@@ -2714,6 +2714,7 @@
     "note_id": "harmony-flag",
     "equals": "chrome62",
     "release": "2018-03-07",
+    "retire": "2019-12-31",
```
Comment (ljharb) [inline environments.json:2717]: generally we don’t put future dates in here; I’d prefer to wait until after something is retire before marking it as such

**Review by ljharb — CHANGES_REQUESTED** [review]

File: environments.json (hunk 1 above)
Comment (zloirock) [reply environments.json:2717]: It's official dates from [Node releases schedule](https://github.com/nodejs/Release/blob/master/schedule.json). Those fields are not used in the generation of tables. So it's just a reminder when they should be marked as obsolete if we'll forget it.

Comment (zloirock) [reply environments.json:2717]: About "generally we don’t put future dates in here" - take a look, for example, at FF ESR.

Comment (ljharb) [reply environments.json:2717]: fair enough, if none of the tools read it then it doesn't matter that it will drift.

**Review by ljharb — APPROVED** [review]

**Pull request merged**